cartographer/mapping/internal/2d/scan_matching/interpolated_tsdf_2d_test.cc

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/mapping/internal/2d/scan_matching/interpolated_tsdf_2d.h"
+
+#include "Eigen/Core"
+#include "gtest/gtest.h"
+
+namespace cartographer {
+namespace mapping {
+namespace scan_matching {
+namespace {
+
+class InterpolatedTSDF2DTest : public ::testing::Test {
+ protected:
+  InterpolatedTSDF2DTest()
+      : tsdf_(MapLimits(1., Eigen::Vector2d(5.5, 5.5), CellLimits(10, 10)),
+              1.0f, 10.0f, &conversion_tables_),
+        interpolated_tsdf_(tsdf_) {}
+
+  float GetUninterpolatedCorrespondenceCost(float x, float y) const {
+    return tsdf_.GetCorrespondenceCost(
+        tsdf_.limits().GetCellIndex(Eigen::Vector2f(x, y)));
+  }
+
+  float GetUninterpolatedWeight(float x, float y) const {
+    return tsdf_.GetWeight(tsdf_.limits().GetCellIndex(Eigen::Vector2f(x, y)));
+  }
+
+  ValueConversionTables conversion_tables_;
+  TSDF2D tsdf_;
+  InterpolatedTSDF2D interpolated_tsdf_;
+};
+
+TEST_F(InterpolatedTSDF2DTest, InterpolatesGridPoints) {
+  std::vector<Eigen::Vector2f> inner_points = {
+      Eigen::Vector2f(1.f, 1.f), Eigen::Vector2f(2.f, 1.f),
+      Eigen::Vector2f(1.f, 2.f), Eigen::Vector2f(2.f, 2.f)};
+  for (const Eigen::Vector2f& point : inner_points) {
+    tsdf_.SetCell(tsdf_.limits().GetCellIndex(point), 0.1f, 1.0f);
+  }
+  // Insert surrounding points.
+  for (size_t x = 0; x < 4; ++x) {
```

Comment:
Might be more readable to list the lines explicitly.

Reply:
Yes.